crates/pathfinder/src/state/merkle_tree.rs

```diff
@@ -463,6 +509,48 @@ impl<T: NodeStorage> MerkleTree<T> {
         Ok(result)
     }
 
+    /// Generates a merkle-proof for every `key` in `keys`.
+    ///
+    /// Returns a vector of vectors of [`ProofNode`] which form a chain from the root to the key,
+    /// if it exists, or upto the node which proves that the key does not exist.
+    ///
+    /// The nodes are returned in order, root first.
+    ///
+    /// Verification is performed by confirming that:
+    ///   1. the chain follows the path of `key`, and
+    ///   2. the hashes are correct, and
+    ///   3. the root hash matches the known root
+    pub fn get_proofs<'a>(
```

Comment:
I would keep this singular and let the caller decide to do multiple calls and how to collect them.

Some benefits of this are:
1. Testing is easier because you don't have to do nested vecs
2. Caller can decide what type of collection he wants. For example, if he wants a `HashMap<StorageAddress, Proof>` he now has to first have a vector and deconstruct it into a hashmap.
3. Caller also does not have to worry that you lied and didn't give `N` results for `N` inputs. Or that you didn't scramble the inputs. So its a way of type safety - by simply not allowing for it.

Reply:
Agreed. Here it is: 5b488972b295aceb65760f711c0bb23a16077582

What do you think about the name of the RPC method though. We have four options:
1. `pathfinder_getProof`
2. `*_getProofs`
3. `*_getStorageProof`
4. `*_getStorageProofs`

knowing that the object we're returning consists of:
`contract_proof` and then `storage_proofs`...

I'm personally thinking that the initial wording, `*_getProof` makes sense, because we're getting storage proofs but it's for a single contract. Idk what's your opinion on the matter.

Reply:
Either `pathfinder_getProof` or `pathfinder_getStorageProof`. I suppose the latter if we are planning on adding proofs for other things 🤔 I can't imagine what that would be though.. I know there are transactions and events that get similar trees built for them, but I can't imagine the use case for proofs of these instead of just getting all of them at once instead.

So I think just the former is fine.

Reply:
pathfinder_getProof would be my 2cents because it might just proof for the account only as well without any proofs for account storage.